hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacitySchedulerQueueManager.java

```diff
@@ -231,99 +229,62 @@ public void reinitializeQueues(CapacitySchedulerConfiguration newConf)
    * @throws IOException
    */
   static CSQueue parseQueue(
```

Comment:
One idea (done it once during YARN-10907, but reverted it when the change was sliced into multiple tickets): AFAIK there is no strong reason why parseQueue is static (apart from saving a few lines of mocking in UTs), but having it static requires two extra parameters (queueContext and conf, both of them are accessible in the class). Maybe we should consider changing it to an instance method. What do you think?

Reply:
I second this idea. However, I think it should be handled in a followup jira instead, because it would involve significant amount of code changes. What is your opinion about this?